flink-streaming-java/src/main/java/org/apache/flink/streaming/api/environment/CheckpointConfig.java

```diff
@@ -154,30 +143,19 @@ public class CheckpointConfig implements java.io.Serializable {
     public CheckpointConfig(final CheckpointConfig checkpointConfig) {
         checkNotNull(checkpointConfig);
 
-        this.checkpointInterval = checkpointConfig.checkpointInterval;
-        this.checkpointingMode = checkpointConfig.checkpointingMode;
-        this.checkpointTimeout = checkpointConfig.checkpointTimeout;
-        this.maxConcurrentCheckpoints = checkpointConfig.maxConcurrentCheckpoints;
-        this.minPauseBetweenCheckpoints = checkpointConfig.minPauseBetweenCheckpoints;
-        this.tolerableCheckpointFailureNumber = checkpointConfig.tolerableCheckpointFailureNumber;
-        this.unalignedCheckpointsEnabled = checkpointConfig.isUnalignedCheckpointsEnabled();
-        this.alignedCheckpointTimeout = checkpointConfig.alignedCheckpointTimeout;
-        this.approximateLocalRecovery = checkpointConfig.isApproximateLocalRecoveryEnabled();
-        this.externalizedCheckpointCleanup = checkpointConfig.externalizedCheckpointCleanup;
-        this.forceCheckpointing = checkpointConfig.forceCheckpointing;
-        this.forceUnalignedCheckpoints = checkpointConfig.forceUnalignedCheckpoints;
+        this.configuration = new Configuration(checkpointConfig.configuration);
         this.storage = checkpointConfig.getCheckpointStorage();
-        this.checkpointIdOfIgnoredInFlightData =
-                checkpointConfig.getCheckpointIdOfIgnoredInFlightData();
     }
 
-    public CheckpointConfig() {}
+    public CheckpointConfig() {
+        configuration = new Configuration();
+    }
 
     // ------------------------------------------------------------------------
 
     /** Disables checkpointing. */
     public void disableCheckpointing() {
-        this.checkpointInterval = -1;
+        configuration.removeConfig(ExecutionCheckpointingOptions.CHECKPOINTING_INTERVAL);
```

Comment:
it would be better to have a set operation here, currently a disabling would be a "fallback to cluster config" if we merge configuration in the future.

Reply:
agrrr, good catch.

Reply:
Even if `CheckpointConfig` is only a view, it is likely that we use `WritableConfig` which has no `removeConfig` contract. At least for now. Every deletion is modelled like a setting right now.

Reply:
I've replaced that with setting to `Duration.ofMillis(-1)` (please check the fixup commit)

Reply:
Not sure if `Duration.ofMillis(-1)` is supported by the entire stack: `org.apache.flink.util.TimeUtils#parseDuration`

Reply:
Also `org.apache.flink.util.TimeUtils#formatWithHighestUnit` could lead to issues if `ExecutionConfig.toConfiguration().toMap`?

Reply:
I guess 0 would be the better option?

Reply:
In the offline discussion we decided to leave it as it was (with the `removeConfig` option), and deal with it later, if it causes a problem.
@twalthr didn't want to change the returned value from `StreamExecutionEnvironment` and `CheckpointConfig` to `0`
I didn't want to have `StreamExecutionEnvironment` and `CheckpointConfig` to return `-1` while `ConfigOption` return `0`
Adding support for negative values in the `ConfigOption` would be problematic, as it would have to be done in such way, that all already existing `ConfigOption` would have to fail/reject negative values.